src/controller/CHIPDeviceController.cpp

```diff
@@ -211,47 +215,90 @@ CHIP_ERROR DeviceController::Init(NodeId localDeviceId, ControllerInitParams par
     return CHIP_NO_ERROR;
 }
 
-CHIP_ERROR DeviceController::LoadLocalCredentials(Transport::AdminPairingInfo * admin)
+CHIP_ERROR DeviceController::GenerateOperationalCertificates(const ByteSpan & CSR, NodeId deviceId, MutableByteSpan & cert)
 {
-    ChipLogProgress(Controller, "Getting operational keys");
-    Crypto::P256Keypair * keypair = admin->GetOperationalKey();
+    // This code requires about 2K RAM to generate the certificates.
+    // The code would run as part of commissioner applications, so RAM requirements should be fine.
+    // Need to analyze if this requirement could be better managed by using static memory pools.
+    chip::Platform::ScopedMemoryBuffer<uint8_t> noc;
```

Comment:
nit: Suggest operationalCredential or nodeOperationalCredential. Abbreviating like this hurts readability (admittedly the spec defines these so it's somewhat justified).